### PR TITLE
Optimize L2 Regularizer

### PR DESCRIPTION
**Pull request by szutenberg**
By using L2Loss op we make graph much smaller.

image

I didn't change docs in order to avoid confusion (math remains unchanged).

### CODE REVIEW

File: keras/regularizers.py [hunk 1]
```diff
@@ -241,7 +241,7 @@ def __call__(self, x):
     if self.l1:
       regularization += self.l1 * tf.reduce_sum(tf.abs(x))
     if self.l2:
-      regularization += self.l2 * tf.reduce_sum(tf.square(x))
+      regularization += 2.0 * self.l2 * tf.nn.l2_loss(x)
```
Comment (fchollet) [inline keras/regularizers.py:244]: IMO `2 * tf.nn.l2_loss(x)` hinders readability compared to `tf.reduce_sum(tf.square(x))`. You don't understand where the 2 comes from and you have to know about an obscure TF API. Basically you cannot understand the code without reading the TF docs.

**Review by fchollet — COMMENTED** [review]
Thanks for the PR.

**Comment by fchollet** [comment]
Generally speaking we should favor code that is written in NumPy-style math APIs whenever possible.

**Comment by szutenberg** [comment]
Hi @fchollet ,

I'm surprised that you don't want to merge this change.

My change is not about improving readability of the code (I fully agree that "2.0" is confusing but that's how it is in TF API), it is performance-related change, similar to https://github.com/tensorflow/tensorflow/pull/47572 (this change is also "against NumPy-style"). 

I added comments in the code: https://github.com/szutenberg/keras/commit/23a9149a8da92767b18f1e0e02de99d60e4e602f.

Could you reconsider it? 

Thanks.

File: keras/regularizers.py (hunk 1 above)
Comment (szutenberg) [reply keras/regularizers.py:244]: You're right. So I added comments in the code: https://github.com/szutenberg/keras/commit/23a9149a8da92767b18f1e0e02de99d60e4e602f.

**Pull request closed without merging**